File: go-gulf-ios-driver-main/GoGulf Driver/Models/Response/NotificationModel.swift
//
//  NotificationModel.swift
//  CTP Limo
//
//  Created by Office on 23/05/2023.
//

import Foundation


// MARK: - Welcome
struct NotificationModel: Codable {
    let title, message: String?
    let data: [NotificationModelData]?
}

// MARK: - Datum
struct NotificationModelData: Codable {
    let id: Int?
    let title, message: String
    let seen: Int?
    let notificationAt: String?

    enum CodingKeys: String, CodingKey {
        case id, title, message, seen
        case notificationAt = "notification_at"
    }
}



struct ReadNotificationModel: Codable{
    let id: Int
}

//struct NotificationListModel: Codable {
//    let bookingId: String?
//    let clientSecret: String?
//    let invoiceId: String?
//    let message: String?
//    let notificationAt: String?
//    let notificationId: Int?
//    let timestamp: Int?
//    let seen: Bool?
//    let target: String?
//    let title: String?
//
//    enum CodingKeys: String, CodingKey {
//        case bookingId = "booking_id"
//        case clientSecret = "client_secret"
//        case invoiceId = "invoice_id"
//        case message
//        case notificationAt = "notification_at"
//        case notificationId = "notification_id"
//        case timestamp
//        case seen
//        case target
//        case title
//    }
//}


struct NotificationCounts: Codable {
    let notification: Int?
}


struct NotificationListModel: Codable, Hashable {
    let id: String?
    let is_seen: Bool?
    let message: String?
    let target: String?
    let timestamp: Int?
    let title: String?
    let type: String?
}

File: go-gulf-andriod-passenger-master/app/src/main/java/com/gogulf/passenger/app/data/model/response/notification/NotificationModel.kt
package com.gogulf.passenger.app.data.model.response.notification

import com.google.firebase.firestore.PropertyName
import com.google.gson.annotations.SerializedName
import java.io.Serializable


data class NoticeModel(
    @set:PropertyName("booking_id")
    @get:PropertyName("booking_id")
    @SerializedName("booking_id")
    var bookingID: String? = "",

    @set:PropertyName("invoice_id")
    @get:PropertyName("invoice_id")
    @SerializedName("invoice_id")
    var invoiceID: String? = "",

    @set:PropertyName("notification_at")
    @get:PropertyName("notification_at")
    @SerializedName("notification_at")
    var notificationAt: String? = "",

    @set:PropertyName("id")
    @get:PropertyName("id")
    @SerializedName("id")
    var notificationId: String? = "",

    var seen: Int? = 0,

    val is_seen: Boolean? = null,
    val message: String? = null,
    val target: String? = null,
    val timestamp: Long? = null,
    val title: String? = null,
    val type: String? = null
) : Serializable